### PR TITLE
refactor(lua): rename the module vscode-neovim to vscode

### PR DESCRIPTION
**Pull request by xiyaowong**
I welcome all feedback! If there's a unanimous agreement that it's unnecessary, I'll close this PR and cease discussing this topic.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -318,9 +318,15 @@ For more information:
...
 
 ```lua
-local vscode = require('vscode-neovim')
+local vscode = require('vscode')
 ```
 
+<!-- prettier-ignore-start -->
+> [!TIP]
+> "vscode" is an alias for "vscode-neovim" module.
+> So don't be confused if "vscode-neovim" is used in the below examples.
```
Comment (theol0403) [inline README.md:327]: Just change the examples. We can change all wording/internals to use `vscode` and have `vscode-neovim` as an undocumented backwards-compatibility.

**Review by justinmk — APPROVED** [review]
This is 100% a good idea. 

vscode-neovim is a very awkward name and should be de-emphasized as much as possible. And there is no risk of conflict with some other "vscode" lua module (we can show an error if we detect such a thing, perhaps)

File: README.md [hunk 2]
```diff
@@ -309,9 +309,15 @@ For more information:
...
```
Comment (justinmk) [inline README.md:318]:
```suggestion
> "vscode" is an alias for "vscode-neovim" (deprecated) module.
> So don't be confused if "vscode-neovim" is used in old discussions or other resources.
```

**Review by justinmk — APPROVED** [review]
🚀  Names matter. Less is more. 🚀

File: README.md (hunk 2 above)
Comment (xiyaowong) [reply README.md:318]: Oh, I forgot to remove this section

**Comment by xiyaowong** [comment]
Obviously more concise and intuitive!

**Review by theol0403 — APPROVED** [review]

**Pull request merged**